src/phoenix/experimental/evals/evaluators.py

```diff
@@ -1,7 +1,107 @@
-from typing import List, Optional
+from dataclasses import dataclass
+from functools import partial
+from typing import Any, List, Mapping, Optional, Protocol
 
-from phoenix.experimental.evals import PromptTemplate
-from phoenix.experimental.evals.models import BaseEvalModel
+from phoenix.experimental.evals.models import set_verbosity
+from phoenix.utilities.logging import printif
+
+from .models import BaseEvalModel
+from .templates import ClassificationTemplate, PromptTemplate
+
+Record = Mapping[str, Any]
+
+NOT_PARSABLE = "NOT_PARSABLE"
+
+
+@dataclass(frozen=True)
+class EvaluationResult:
+    """
+    A class to contain the results of an evaluation.
+    """
+
+    prediction: str
+
+
+class Evaluator(Protocol):
```

Comment:
If we are doing duck typing, then protocol should live with the function that uses it because it's an input type, and not with the implementations themselves (otherwise it could just be inheritance) because the key purpose of duck typing (compared to inheritance) is to sever the explicit dependency (but whether that severance is necessary is a separate discussion).

Reply:
I'm removing until we have a concrete need for other kinds of evaluators. Addressed in f8689fbe